QuizApp/PurpleBackgroundLayerProtocol.swift

```diff
@@ -0,0 +1,31 @@
+import UIKit
+
+protocol PurpleBackgroundLayerProtocol {
```

Comment:
```suggestion
protocol PurpleViewController: UIViewController {
```

Because you know you will always use this protocol on a view controller, you could make your protocol inherit UIViewController. That will also limit protocol adoption to UIViewController class types, which is why in that case I suggest renaming the protocol to PurpleViewController so that it is clear from the name. 

What you will gain in that case is that your `createPurpleBackgroundLayer(view: UIView)` will not need that view argument anymore, because it will already have a viewController's view visible inside.